macros/src/main/scala/com/gu/contentapi/circe/CirceScroogeMacros.scala

```diff
@@ -187,4 +281,49 @@ object CirceScroogeMacros {
     tree
 
   }
+
+  implicit def encodeThriftUnion[A <: ThriftUnion]: Encoder[A] = macro encodeThriftUnion_impl[A]
+
+  def encodeThriftUnion_impl[A: c.WeakTypeTag](c: blackbox.Context): c.Tree = {
+    import c.universe._
+
+    val A = weakTypeOf[A]
+
+    val memberClasses: Iterable[Symbol] = A.companion.members.filter { member =>
```

Comment:
By using macro bundles (http://docs.scala-lang.org/overviews/macros/bundles), you can have multiple methods sharing the same context/universe, which allows you to factor out common methods like this.